test/widgets/store_test.dart

```diff
@@ -113,4 +150,65 @@ void main() {
     check(tester.any(find.textContaining('found store'))).isTrue();
     tester.widget(find.text('found store, account: ${eg.selfAccount.id}'));
   });
+
+  testWidgets('PerAccountStoreAwareStateMixin', (tester) async {
+    final widgetWithMixinKey = GlobalKey<MyWidgetWithMixinState>();
+    final accountId = eg.selfAccount.id;
+
+    await TestZulipBinding.instance.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+
+    Future<void> pumpWithParams({required bool light, required int accountId}) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          theme: light ? ThemeData.light() : ThemeData.dark(),
+          home: GlobalStoreWidget(
+            child: PerAccountStoreWidget(
+              accountId: accountId,
+              child: MyWidgetWithMixin(key: widgetWithMixinKey))),
+        ));
+    }
+
+    // [onNewStore] called initially
+    await pumpWithParams(light: true, accountId: accountId);
+    await tester.pump(); // global store
+    await tester.pump(); // per-account store
+    check(widgetWithMixinKey).currentState.isNotNull().storeChangeCounter.equals(1);
+    check(widgetWithMixinKey).currentState.isNotNull()
+      ..anyDepChangeCounter.equals(1)
+      ..storeChangeCounter.equals(1);
+
+    // [onNewStore] not called on unrelated dependency change
+    await pumpWithParams(light: false, accountId: accountId);
+    await tester.pumpAndSettle();
+    check(widgetWithMixinKey).currentState.isNotNull()
+      ..anyDepChangeCounter.equals(3) // (Dunno why 3 instead of 2. Anyway, more than 1…)
+      ..storeChangeCounter.equals(1);
+
+    // [onNewStore] called when store changes
+    //
+    // TODO: Trigger `store` change by simulating an event-queue renewal,
+    //   instead of with this hack where we change the UI's backing Account
+    //   out from under it. (That account-swapping would be suspicious in
+    //   production code, where we could reasonably add an assert against it.
+    //   If forced, we could let this test code proceed despite such an assert…)
+    // hack; the snapshot probably corresponds to selfAccount, not otherAccount.
+    await TestZulipBinding.instance.globalStore.add(eg.otherAccount, eg.initialSnapshot());
+    await pumpWithParams(light: false, accountId: eg.otherAccount.id);
+    // Nudge PerAccountStoreWidget to send its updated store to MyWidgetWithMixin.
+    //
+    // A change in PerAccountStoreWidget's [accountId] field doesn't by itself
+    // prompt dependant widgets (those using PerAccountStoreWidget.of) to update,
+    // even though it holds a new store. (See its state's didUpdateWidget,
+    // or lack thereof.) That's reasonable, since such a change is never expected;
+    // see TODO above.
+    //
+    // But it will prompt widgets to update when it receives a notification
+    // from the GlobalStore; see its state's didChangeDependencies.
+    // So, take advantage of that.
```

Comment:
See what you think of the comment in my next revision; I'm not sure I've found the right balance of explaining and being brief. 🙂

Reply:
Looks fine. :-) Might be possible to make it briefer; but the other thing to balance is effort spent, so for this context this new version is good. (If it were some central API's doc rather than an implementation comment in a test, I'd probably give it another pass.)